tools/workflow/clean_aws_rsc.go

```diff
@@ -0,0 +1,167 @@
+			}))
+
+	ec2client := ec2.New(testSession)
+
+	// Get list of instance
+	//State filter
+	instanceStateFilter := ec2.Filter{Name: aws.String("instance-state-name"), Values: []*string{aws.String("running")}}
+
+	//Tag filter
+	instanceTagFilter := ec2.Filter{Name: aws.String("tag:Name"), Values: []*string{aws.String("Integ-test-Sample-App"), aws.String("Integ-test-aoc")}}
+
+	//get instances to delete
+	var deleteInstanceIds []*string
+	var nextToken *string
+	for {
+		describeInstancesInput := ec2.DescribeInstancesInput{Filters: []*ec2.Filter{&instanceStateFilter, &instanceTagFilter}, NextToken: nextToken}
+		describeInstancesOutput, describeInstanceErr := ec2client.DescribeInstances(&describeInstancesInput)
+		if describeInstanceErr != nil {
+			log.Printf("Failed to get instance for error %v", describeInstanceErr)
+			return describeInstanceErr
+		}
+		for _, reservation := range describeInstancesOutput.Reservations {
+			for _, instance := range reservation.Instances {
+				//only delete instance if older than 5 days
+				if time.Now().UTC().Add(-1 * time.Hour * 24 * pastDayDelete).After(*instance.LaunchTime) {
+					log.Printf("Try to delete instance %s tags %v launch-date %v", *instance.InstanceId, instance.Tags, instance.LaunchTime)
+					deleteInstanceIds = append(deleteInstanceIds, instance.InstanceId)
+				}
+			}
+		}
+		if describeInstancesOutput.NextToken == nil {
+			break
+		}
+		nextToken = describeInstancesOutput.NextToken
+	}
+
+	if len(deleteInstanceIds) < 1 {
+		log.Printf("No instances to delete")
+		return nil
+	}
+
+	terminateInstancesInput := ec2.TerminateInstancesInput{InstanceIds: deleteInstanceIds}
+	_, terminateInstancesErr := ec2client.TerminateInstances(&terminateInstancesInput)
+	if terminateInstancesErr != nil {
+		log.Printf("Failed to terminate instances %v because of %v",deleteInstanceIds, terminateInstancesErr)
+		return terminateInstancesErr
+	}
+
+	return nil
+}
+
+func destroyLoadBalancerResource()(error){
+	// Set up aws go sdk session
+	//Enable default region and credentials
+	//Documents: https://docs.aws.amazon.com/ja_jp/sdk-for-go/v1/developer-guide/configuring-sdk.html
+	session := session.Must(
+		session.NewSessionWithOptions(
+			session.Options{
+				SharedConfigState: session.SharedConfigEnable,
+			}))
+
+	svc := elbv2.New(session)
+
+	//ELB Go SDK currently does not support filter tag or filter wildcard. Only supports with matching name
+	//Documentation: https://github.com/aws/aws-sdk-go/blob/02266ed24221ac21bb37d6ac614d1ced95407556/service/elbv2/api.go#L5879-L5895
+
+	describeLoadBalancerInputs := &elbv2.DescribeLoadBalancersInput{}
+
+
+	describeLoadBalancerOutputs, describeLoadBalancerErr := svc.DescribeLoadBalancers(describeLoadBalancerInputs)
+
+	if describeLoadBalancerErr != nil {
+		log.Printf("Failed to get metadata for load balancer because of %v", describeLoadBalancerErr)
+		return describeLoadBalancerErr
+	}
+
+	for _, lb := range describeLoadBalancerOutputs.LoadBalancers {
```

Comment:
Will we need to take into account pagination? Probs not, since this will run on a timer and eventually we will end up with less than the number give for describe.

Reply:
Yes. You are right. The elbv2 only supports application and network lb, excludes the classic one. That's why it's hard for me to imagine these elbv2 can exceed 50 which is the default for paginate. However, adding next Marker to prevent that in the long run.